ckanext/canada/tests/test_search.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: UTF-8 -*-
+from ckanapi import LocalCKAN
+import ckan.plugins as p
+
+from ckan.tests.helpers import FunctionalTestBase
+from ckan.tests.factories import Sysadmin
+from ckanext.canada.tests.factories import (
+    CanadaOrganization as Organization,
+    CanadaUser as User,
+    CanadaDataset as Dataset
+)
+
+
+class TestRegistrySearch(FunctionalTestBase):
+    """
+    Class to test the package_search functionality for the Registry.
+    """
+    def setup(self):
+        super(TestRegistrySearch, self).setup()
+        # all datasets in canada_internal are private
+        self.include_private = True
+        user = User()
+        editor = User()
+        sysadmin = Sysadmin()
+        self.org = Organization(users=[
+            {'name': editor['name'],
+             'capacity': 'editor'},
+            {'name': sysadmin['name'],
+             'capacity': 'admin'}])
+        self.user_lc = LocalCKAN(username=user['name'])
+        self.editor_lc = LocalCKAN(username=editor['name'])
+        self.system_lc = LocalCKAN(username=sysadmin['name'])
+        #
+        # Datasets 0, 1, 2, and 3
+        # 1 & 3 will be ready to publish
+        # 3 will not have a portal release date
+        # datasets belong to self.org
+        #
+        for _i in range(0, 4):  # 0-3
```

Comment:
`i` is a common loop variable, why `_i`?

Reply:
@wardi I just personally use `_<var name>` for local scope. And the `_` before the variable in the loop will ignore any typing for the basic loop.

Reply:
in Python `_<var name>` means a name that shouldn't be imported from outside modules, so it's kind of funny looking. What do you mean about ignoring typing?

Reply:
All variables in python are local unless you don't declare them or use `nonlocal foo` or `global foo` in the function, and there is a convention of *not* prefixing variable names for scope, type or anything else. There are some conventions around capitalization, however.